### PR TITLE
transport: implement ELASTIC_APM_SERVER_CERT

### LINKED ISSUE #312: Implement support for APM Server certificate pinning
Primarily for self-signed certificates.

### PR DESCRIPTION
**Pull request by axw**
Implement server certificate pinning.

- [x] implement HTTP transport changes
- [x] docs
- [x] live test with APM server

Closes elastic/apm-agent-go#312

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -315,6 +315,24 @@ To reduce overhead and storage requirements, you can set the sample rate to a va
...
+| `ELASTIC_APM_SERVER_CERT`  |
+|============
+
+If you have configured your APM Server with a self signed TLS certificate, you can
+specify the path to the certificate via the `ELASTIC_APM_SERVER_CERT` configuration.
+This will enable the agent to verify the server's authenticity.
+
+By default, the agent verifies the server's certificate if you use an
+HTTPS connection to the APM server. Verification can be disabled by
+changing this setting to `false`.
```
Comment (roncohen) [inline docs/configuration.asciidoc]: why can you set this to `false` ? Copy/paste error?

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -315,6 +315,24 @@ To reduce overhead and storage requirements, you can set the sample rate to a va
...
+[[config-server-cert]]
+=== `ELASTIC_APM_SERVER_CERT`
+
+[options="header"]
+|============
+| Environment                | Default
+| `ELASTIC_APM_SERVER_CERT`  |
+|============
+
+If you have configured your APM Server with a self signed TLS certificate, you can
```
Comment (roncohen) [inline docs/configuration.asciidoc]: would work even for an actual CA certified certificate right? I get that the primary use-case is self signed, but i've seen people pin them anyway. I don't feel strongly about rewording this, so feel free to leave it as is.

Comment (roncohen) [reply docs/configuration.asciidoc]: I like that it says PEM encoded [further down](https://github.com/elastic/apm-agent-go/pull/325/files#diff-2d08121c527972df0fc40641b3ecc1a2R72). Might want to include that here as well.

File: transport/http.go [hunk 3]
```diff
@@ -278,3 +299,34 @@ func requestWithContext(ctx context.Context, req *http.Request) *http.Request {
...
+		if certBlock == nil {
+			return nil, errors.New("missing or invalid certificate")
+		}
+		if certBlock.Type == "CERTIFICATE" {
+			return x509.ParseCertificate(certBlock.Bytes)
+		}
+	}
+}
+
+func verifyPeerCertificate(rawCerts [][]byte, trusted *x509.Certificate) error {
```
Comment (roncohen) [inline transport/http.go]: I have no idea if this is how it's done. Any pointers?

Comment (roncohen) [reply transport/http.go]: OK, LGTM

File: transport/http.go [hunk 4]
```diff
@@ -26,6 +28,7 @@ const (
 	envServerURLs       = "ELASTIC_APM_SERVER_URLS"
 	envServerURL        = "ELASTIC_APM_SERVER_URL"
 	envServerTimeout    = "ELASTIC_APM_SERVER_TIMEOUT"
+	envServerCert       = "ELASTIC_APM_SERVER_CERT"
```
Comment (roncohen) [inline transport/http.go]: Should probably be `ELASTIC_APM_SERVER_CERT`?

File: transport/http_test.go [hunk 5]
```diff
@@ -29,6 +30,7 @@ func init() {
 	os.Unsetenv("ELASTIC_APM_SERVER_URLS")
 	os.Unsetenv("ELASTIC_APM_SERVER_URL")
 	os.Unsetenv("ELASTIC_APM_SECRET_TOKEN")
+	os.Unsetenv("ELASTIC_APM_SECRET_CERT")
```
Comment (roncohen) [inline transport/http_test.go]: should probably be `ELASTIC_APM_SERVER_CERT`?

File: docs/configuration.asciidoc (hunk 1 above)
Comment (axw) [reply docs/configuration.asciidoc]: copy pasta from the config below, thanks

File: transport/http_test.go (hunk 5 above)
Comment (axw) [reply transport/http_test.go]: oops, thanks!

**Pull request merged**